components/tools/OmeroPy/src/omero/util/script_utils.py

```diff
@@ -1234,3 +1230,34 @@ def findROIByImage(roiService, image, namespace):
     for roi in results.rois:
         roiList.append(ROIData(roi))
     return roiList
+
+
+def numpyToImage(plane, minMax, type):
+    """
+    Converts the numpy plane to a PIL Image, converting data type if necessary.
+    @param plane The plane to handle
+    @param minMax the min and the max values for the plane
+    @param type the data type to use for scaling
+    """
+
+    import numpy
+    from numpy import add, zeros
+    try:
+        from PIL import Image
+    except ImportError:
+        import Image
+
+    if plane.dtype.name not in ('uint8', 'int8'):   # we need to scale...
+        minVal, maxVal = minMax
+        valRange = maxVal - minVal
+        if (valRange <= 0):
+            valRange = 1
+        scaled = (plane - minVal) * (float(255) / valRange)
+        convArray = zeros(plane.shape, dtype=type)
+        if numpy.__version__ > (1, 7):
```

Comment:
I am not convince about that logic

Tested with various versions and only noticed failure on:

```
Traceback (most recent call last):
  File "/Users/ola/OMERO/DEV/TEST/OMERO.py/lib/python/omero/util/script_utils.py", line 1260, in numpyToImage
    convArray += scaled
TypeError: Cannot cast ufunc add output from dtype('float64') to dtype('int32') with casting rule 'same_kind'
```

how about to change it to:

```
        try:
            convArray += scaled
        except:
            add(convArray, scaled, out=convArray, casting="unsafe")
```